libsolutil/Exceptions.h

```diff
@@ -76,6 +76,7 @@ DEV_SIMPLE_EXCEPTION(FileNotFound);
 DEV_SIMPLE_EXCEPTION(NotAFile);
 DEV_SIMPLE_EXCEPTION(DataTooLong);
 DEV_SIMPLE_EXCEPTION(StringTooLong);
+DEV_SIMPLE_EXCEPTION(ElementNotFound);
```

Comment:
I just added a simple exception, but maybe a more specific one would be better.

Reply:
Right! I've added the asserts to validate indexes bounds and removed the exceptions. I've also removed the tests related  to the exceptions. Should I instead replace them with tests that check for a runtime error in case these assertions fail?

Reply:
No, we don't test assertions. We assume we will never write code that uses indexes out of bounds and assertions are there as a safety net in case we accidentally do.